WordPress/Classes/Networking/PlansRemote.swift

```diff
@@ -54,9 +56,162 @@ private func mapPlansResponse(response: AnyObject) throws -> (activePlan: Plan,
+        guard let slugs = groupJson["items"] as? [String] else { throw PlansRemote.Error.DecodeError }
+        return PlanFeatureGroup(title: groupJson["title"] as? String, slugs: slugs)
+    }
+    
+    PlanFeatureGroup.setGroups(groups, forPlan: plan)
+}
+
+class PlanFeaturesRemote: ServiceRemoteREST {
+    
+    enum Error: ErrorType {
+        case DecodeError
+    }
+    
+    private var cacheDate: NSDate?
+    private let cacheFilename = "plan-features.json"
+    
+    func getPlanFeatures(success: PlanFeatures -> Void, failure: ErrorType -> Void) {
+        // First we'll try and return plan features from memory, then check our disk cache,
+        // and finally hit the network if we don't have anything recent enough
+        if let planFeatures = inMemoryPlanFeatures {
+            success(planFeatures)
+            return
+        }
+        
+        // If we have features cached to disk, update the in-memory list and the cache date to match
+        if let (planFeatures, date) = cachedPlanFeaturesWithDate() {
+            inMemoryPlanFeatures = planFeatures
+            cacheDate = date
+            
+            success(planFeatures)
+            return
+        }
+        
+        fetchPlanFeatures({ [weak self] planFeatures in
+            self?.inMemoryPlanFeatures = planFeatures
+            self?.cacheDate = NSDate()
+            
+            success(planFeatures)
+        }, failure: failure)
+    }
+    
+    private var _planFeatures: PlanFeatures?
+    private var inMemoryPlanFeatures: PlanFeatures? {
+        get {
+            // If we have something in memory and it's less than a day old, return it.
+            if let planFeatures = _planFeatures,
+                let cacheDate = cacheDate where
+                cacheDateIsValid(cacheDate) {
+                    return planFeatures
+            }
+            
+            return nil
+        }
+        
+        set {
+            _planFeatures = newValue
+        }
+    }
+    
+    private func cachedPlanFeatures() -> PlanFeatures? {
+        guard let (planFeatures, _) = cachedPlanFeaturesWithDate() else { return nil}
+
+        return planFeatures
+    }
+    
+    /// - returns: An optional tuple containing a collection of cached plan features and the date when they were fetched
+    private func cachedPlanFeaturesWithDate() -> (PlanFeatures, NSDate)? {
+        guard let cacheFileURL = cacheFileURL,
+            let path = cacheFileURL.path,
+            let attributes = try? NSFileManager.defaultManager().attributesOfItemAtPath(path),
+            let modificationDate = attributes[NSFileModificationDate] as? NSDate,
+            let response = NSData(contentsOfURL: cacheFileURL),
+            let json = try? NSJSONSerialization.JSONObjectWithData(response, options: []),
+            let planFeatures = try? mapPlanFeaturesResponse(json)
+            where cacheDateIsValid(modificationDate) else { return nil }
```

Comment:
Maybe split this in two `guard`s? If `cacheDateIsValid` returns false, we'd have read the cache from disk, converted it to JSON, and then parsed it, just to return a `nil`.